gobblin-runtime/src/main/java/org/apache/gobblin/runtime/FsDatasetStateStore.java

```diff
@@ -236,6 +236,10 @@ public String sanitizeDatasetStatestoreNameFromDatasetURN(String storeName, Stri
     WritableShimSerialization.addToHadoopConfiguration(deserializeConfig);
     try (@SuppressWarnings("deprecation") SequenceFile.Reader reader = new SequenceFile.Reader(this.fs, tablePath,
         deserializeConfig)) {
+
+      if (reader.getValueClass() != JobState.class && reader.getValueClass() != JobState.DatasetState.class ){
```

Comment:
Maybe instead of doing this check we can use the existing matching classes. We may be able to remove the shims if that works.